packages/client/components/DashNavList/LeftNavPageLink.tsx

```diff
@@ -85,14 +109,32 @@ export const LeftNavPageLink = (props: Props) => {
   const isSelf = draggingPageId === id
   const isNextPeer = draggingPageId === nextPeerId && !showChildren
   const isPrivateToTopLevelShared = isTopLevelShared && draggingPageIsPrivate
-  const canDropIn = draggingPageId && !isSourceDragParent && !isSelf && !isDraggingLastChild
+  const isViewerOwnerOfDraggingPage = draggingPageViewerAccess === 'owner'
+  const dropInSection = `User_pages:${id}`
+  const dropBelowSection =
+    connectionKey === 'User_pages' ? `${connectionKey}:${parentPageId || teamId}` : connectionKey
+
+  const isDropInReorder = draggingPageParentSection === dropInSection
+  const isDropBelowReorder = draggingPageParentSection === dropBelowSection
+  const hasDragAccess = isViewerOwnerOfDraggingPage || isDropInReorder
+  const hasDragDropInAccess = hasDragAccess && isViewerPageEditor
+  const isEditorOfDroppingSection =
+    connectionKey !== 'User_pages' ? true : teamId ? true : isViewerParentPageEditor
+  const hasDragDropBelowAccess =
+    isDropBelowReorder || (isViewerOwnerOfDraggingPage && isEditorOfDroppingSection)
+
+  const canDropIn =
+    draggingPageId && !isSourceDragParent && !isSelf && !isDraggingLastChild && hasDragDropInAccess
+  const hasDragDropInOrBelow = showChildren ? hasDragDropInAccess : hasDragDropBelowAccess
```

Comment:
@Dschoordsch i absolutely hate this spaghetti code.  I've tried refactoring but i just end up with the same convoluted logic. If you see a simple path, please let me know!